fauxfactory/__init__.py

```diff
@@ -520,25 +520,50 @@ def gen_negative_integer():
     return gen_integer(max_value=max_value)
 
 
-def gen_ipaddr(ip3=False, ipv6=False):
+def gen_ipaddr(ip3=False, ipv6=False, prefix=None):
     """Generates a random IP address.
 
+    You can also specify an IP address prefix if you are interested in
+    local network address generation, etc.
+
     :param bool ip3: Whether to generate a 3 or 4 group IP.
     :param bool ipv6: Whether to generate IPv6 or IPv4
+    :param list prefix: A prefix to be used for IPv4 (like [10, 0, 1])
     :returns: An IP address.
     :rtype: str
+    :raises: ``ValueError`` if ``prefix`` is too long either to fit into
+        the IP address or disabling randomness at all.
 
     """
 
+    # Set the lengths of the randomly generated sections
+    if ipv6:
+        rng = 8
+    elif ip3:
+        rng = 3
+    else:
+        rng = 4
+    if not isinstance(prefix, (list, tuple)) and prefix is not None:
+        raise ValueError("Prefix must be a list, tuple, or None, not {}".format(
+            type(prefix).__name__))
+    prefix = [str(field) for field in prefix or []]
+    rng -= len(prefix)
```

Comment:
Add a comment as to what's happening here? I know that you're subtracting the length of prefix from the `rng` so that you know how many random values to generate, but it would be nice to be explicit about it so that it is easier to follow the code later on :)